realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -861,50 +932,56 @@ void BaseRealSenseNode::publishFrame(rs2::frame f, const rclcpp::Time& t,
         image = fix_depth_scale(image, _depth_scaled_image[stream]);
     }
 
-    if (info_publishers.find(stream) == info_publishers.end() ||
-        image_publishers.find(stream) == image_publishers.end())
+    bool is_sc_stream = isSCStream(stream);
+
+    // if stream is on, and is not a SC stream, publish camereaInfo
+    if(!is_sc_stream && info_publishers.find(stream) != info_publishers.end())
```

Comment:
I would have renamed the is_sc_stream to should_publish_camera_info, since this is the aim of it

Reply:
changed the function name from isSCStream to shouldPublishCameraInfo, and changed the implementation inside it.